### DISCUSSION TITLE
Protocol conversion for quip:// and webexstart://

### DISCUSSION THREAD

**Discussion by AshKash** [johnste/finicky#226] [General]
This is a great utility that I'm finding new uses every day. At work, we use custom install with custom domain for both Quip and Webex. One of the biggest problems is we tend to have hundreds of tabs open in Safari. I know you can configure to rewrite urls and convert the protocols. However, I've had no success doing this either for Quip or for Webex.

The Quip url looks like this: `https://quip-apple.com/AzY7Aljs8fzu` but just changing `https` to `quip` doesn't work - it just activates the app. 

Same issue with Webex, except the url is something like: `webexstart://launch/<double base64encoded blob>`. It looks like the server generates this based on one of the url parameters. 

Does anyone have any ideas how I can reverse engineer these? Is there any documentation on Quip or Webex deep linking schemes?

**Comment by johnste** [comment]
Hey!

I'm not a user of either service so I can't try this out on my own, but what I would do, assuming there are landing pages that create the `quip` and the `webexstart`, is to try and reverse engineer the page itself. If you're lucky, it's a simple process you can do figure out without digging down deep in the code.

**Comment by waider** [comment]
Try this:
```
        match: finicky.matchHostnames( [ "quip-apple.com" ] ),
        url: ( {
            url
        } ) => ( {
            ...url,
            host: "",
            search: "",
            pathname: url.pathname.split( '/' )[ 1 ] == "email" ? decodeURIComponent(url.search).split('/')[2].split('&')[0] : url.pathname.split( '/' )[ 1 ],
            protocol: "quip"
        } )
/////////
        {
            match: ( {
                url
            } ) => url.protocol === "quip",
            browser: "Quip"
        }
```

I have noticed that when using Finicky to handle Quip URLs, however, that the Quip application moves to the current desktop when you open a URL, instead of the current desktop flipping to wherever the Quip app is. @johnste this seems unique to the Quip app, but my config to date only uses Chrome and  Chime as its other non-default browsers - and they work as expected, flipping the desktop rather than moving -  so I have a limited set of things to test with.

**Comment by wydwww** [comment]
Probably a similar question:
How to config to open a twitter URL in a browser, instead of Twitter desktop app?

For example, opening any twitter URL (like https://twitter.com/JeffDean/status/1281076033018458112) will call the Twitter desktop app, although the URL test in the Finicky console said this twitter URL should be opened by a browser.
I suppose this is about the custom URL scheme, and I checked the system configuration at `~/Library/Preferences/com.apple.LaunchServices/com.apple.launchservices.secure.plist`, but it is really messy.

Is there a way to config Finicky to avoid the desktop app? Thanks.
